### DISCUSSION TITLE
Resolving jaxlib conflicts with CUDA11.2

### DISCUSSION THREAD

**Discussion by joshdorrington** [jax-ml/jax#22615] [Q&A]
Hi,
I am trying to run [neuralgcm](https://github.com/google-research/neuralgcm) on my local cluster, which is unfortunately limited to CUDA 11.2, and its been suggested I ask over here for some help resolving package conflicts, if even possible.

The main issue is that neuralgcm requires Python>=3.10 and seemingly jaxlib>=4.27, but when I look here:

https://storage.googleapis.com/jax-releases/jax_cuda_releases.html

I can't find a wheel that supports Python>=3.10, jaxlib>=4.27, and CUDA<=11.2

Am I missing any tricks here?

Details about my GPUs below, my effort to build a working python environment is attached, as is the code I'm trying to run. In the current [min.txt](https://github.com/user-attachments/files/16359408/min.txt), it doesn't make it past the jax import. In previous attempts it makes it to the end of the script but then fails due to CUDA and CUDNN incompatibilities.
_______________________________________
Running nvidia-smi in terminal:
`        NVIDIA-SMI 460.106.00   Driver Version: 460.106.00   CUDA Version: 11.2     | `
[environment_configuration.txt](https://github.com/user-attachments/files/16359217/environment_configuration.txt)
[min.txt](https://github.com/user-attachments/files/16359410/min.txt)

**Comment by jakevdp** [comment]
Newer versions of JAX do not support CUDA 11.2. See for example the JAX [Change log](https://jax.readthedocs.io/en/latest/changelog.html#jax-0-4-8-march-29-2023), which mentions that support for CUDA 11.4 and below was dropped in JAX v0.4.8.

Unfortunately, I think your options here are either to update CUDA on your cluster, or limit yourself to older versions of JAX and other packages.
